iothub/device/src/MessageAcknowledgement.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// The options for acknowledging a cloud-to-device (C2D) message.
+    /// </summary>
+    /// <seealso href="https://learn.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-c2d#the-cloud-to-device-message-life-cycle"/>
+    public enum CloudToDeviceMessageAcknowledgement
+    {
+        /// <summary>
+        /// The message will be positively acknowledged.
+        /// </summary>
+        /// <remarks>
+        /// This removes the message from the queue; it will not be sent again.
+        /// </remarks>
+        Complete,
+
+        /// <summary>
+        /// The message will be re-queued to be sent again.
+        /// </summary>
+        /// <remarks>
+        /// This option is not supported over MQTT.
+        /// </remarks>
+        Abandon,
+
+        /// <summary>
+        /// The message will be negatively acknowledged.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This removes the message from the queue; it will not be sent again.
+        /// </para>
+        /// <para>
+        /// This option is not supported over MQTT.
```

Comment:
What will the SDK do if a user returns it? Ignore and send Complete anyway?

Reply:
Yes, and log an error saying "not supported"